lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -1104,8 +1104,6 @@ def _apply_label_matcher(self, expression, topology):
             return self._exec(args)
         except Exception as e:
             logger.debug('Applying the expression failed: "{}", falling back to the original', e)
```

Comment:
Is it possible to catch a more specific exception?

Reply:
good question! regardless of what happened, we want it to catch and return the original, so I think it's appropriate in this case.

Reply:
`_exec` can raise the combination of `subprocess.run` and `result.stdout.decode` which is quite a bit, and I guess it doesn't matter why it failed.

Reply:
`decode` is really unlikely to fail unless the LOCALE has been changed inside a pod somehow.

```python
try:
  subprocess.run(...
except subprocess.CalledProcessError as e:
  # do something
```

Is probably a sufficient guard. I'm more surprised that the linters aren't yelling about the bare `Exception`

Reply:
```
# noinspection PyBroadException
```

they would have, but I've suppressed them for that part as it, as I said, is intentional. 👍🏽